src/mlpack/core/optimizers/smorms3/smorms3.hpp

```diff
@@ -0,0 +1,153 @@
+/**
+ * @file smorms3.hpp
+ * @author Vivek Pal
+ *
+ * SMORMS3 i.e. squared mean over root mean squared cubed optimizer. It is a
+ * hybrid of RMSprop, which estimates a safe and optimal distance based on
+ * curvature and Yann LeCun’s method in "No more pesky learning rates".
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef __MLPACK_CORE_OPTIMIZERS_SMORMS3_SMORMS3_HPP
```

Comment:
We shouldn't use __ as a prefix to the macros, just MLPACK_CORE_OPTIMIZERS_SMORMS3_SMORMS3_HPP will work fine. (That applies to the other files you've added too.)